### PR TITLE
switch enable-tests to enable_tests

### PR DESCRIPTION
**Pull request by vtorri**

### CODE REVIEW

**Comment by kinichiro** [comment]
I had confirmed that `./configure --disable-tests` disabled building and running regression tests.

**Comment by kinichiro** [comment]
Sorry, misunderstood. This defaults to disable testing.

File: configure.ac [hunk 1]
```diff
@@ -66,6 +66,10 @@ AC_ARG_ENABLE([extratests],
 	AS_HELP_STRING([--enable-extratests], [Enable extra tests that may be unreliable on some platforms]))
 AM_CONDITIONAL([ENABLE_EXTRATESTS], [test "x$enable_extratests" = xyes])
 
+AC_ARG_ENABLE([tests],
+	AS_HELP_STRING([--enable-tests], [Enable tests]))
```
Comment (kinichiro) [inline configure.ac:70]: I think enable_tests should be default to yes.

```
AC_ARG_ENABLE([tests],
        AS_HELP_STRING([--enable-tests], [Enable tests]),
        [],
        [enable_tests=yes])
```

**Comment by vtorri** [comment]
i'm right now improving it, thanks

File: configure.ac (hunk 1 above)
Comment (vtorri) [reply configure.ac]: it should be good now

File: configure.ac [hunk 2]
```diff
@@ -66,6 +66,16 @@ AC_ARG_ENABLE([extratests],
 	AS_HELP_STRING([--enable-extratests], [Enable extra tests that may be unreliable on some platforms]))
 AM_CONDITIONAL([ENABLE_EXTRATESTS], [test "x$enable_extratests" = xyes])
 
+AC_ARG_ENABLE([tests],
+       [AS_HELP_STRING([--disable-tests], [Enable tests (default: yes)])],
```
Comment (kinichiro) [inline configure.ac:70]: If using "--disable-tests", help message would be "Disable tests" ?
IMHO, default value description is not needed like "--disable-asm".

File: configure.ac [hunk 3]
```diff
@@ -66,6 +66,16 @@ AC_ARG_ENABLE([extratests],
...
+       [if test "x$enable-tests" = "xyes"; then
```
Comment (kinichiro) [inline configure.ac:71]: s/enable-tests/enable_tests/
This causes "--enable-tests", "--enable-tests=yes", "--enable-tests=no" gives disabling tests.

Comment (kinichiro) [reply configure.ac:71]: And I wonder if we could do this only with "enable_tests" without "have_tests" to keep simple.

File: configure.ac [hunk 4]
```diff
@@ -66,6 +66,15 @@ AC_ARG_ENABLE([extratests],
...
 
+AC_ARG_ENABLE([tests],
+       [AS_HELP_STRING([--disable-tests], [Disable tests @<:@default=yes@:>@])],
+       [
+        if ! test "x${enable_tests}" = "xyes"; then
+		enable_tests="no"
+	fi],
+       [have_tests="yes"])
+AM_CONDITIONAL([ENABLE_TESTS], [test "x$have_tests" = xyes])
+
```
Comment (kinichiro) [inline configure.ac:77]: This does not work properly when using "--enable-tests" and "--enable-tests=yes", and it disables tests.
Help string shows only "--disable-tests" though, I think "--enable-tests" also has to work correctly.
This uses 2 variables "enable_tests" and "have_tests", but I believe we can handle this situation by "enable_tests" only.

Changes below could handle all of "--disable-tests", "--enable-tests", "--enable-tests=yes", "--enable-tests=no" properly.

```
AC_ARG_ENABLE([tests],
       AS_HELP_STRING([--disable-tests], [Disable tests]),
       [],
       [enable_tests=yes])
AM_CONDITIONAL([ENABLE_TESTS], [test "x$enable_tests" = xyes])
```

Can you try this ?

Comment (vtorri) [reply configure.ac:77]: for the help, why adding help for --enable-tests if tests are enabled by default ? Packagers are not stupid

Comment (kinichiro) [reply configure.ac:77]: I did not mean that we have to add help string for "--enable-tests".
I wanted to say "--enable-tests" option has to work correctly as same as "--disable-tests".

And Your last commit appears to work right.

**Pull request merged**